omnigibson/scenes/scene_base.py

```diff
@@ -58,25 +78,56 @@ def __init__(
             floor_plane_visible (bool): whether to render the additionally added floor plane
             floor_plane_color (3-array): if @floor_plane_visible is True, this determines the (R,G,B) color assigned
                 to the generated floor plane
+            use_skybox (bool): whether to load a skybox into the simulator
         """
         # Store internal variables
         self.scene_file = scene_file
         self._loaded = False  # Whether this scene exists in the stage or not
         self._initialized = False  # Whether this scene has its internal handles / info initialized or not (occurs AFTER and INDEPENDENTLY from loading!)
         self._registry = None
-        self._world_prim = None
+        self._scene_prim = None
         self._initial_state = None
         self._objects_info = None  # Information associated with this scene
         self._use_floor_plane = use_floor_plane
         self._floor_plane_visible = floor_plane_visible
         self._floor_plane_color = floor_plane_color
-        self._floor_plane = None
         self._use_skybox = use_skybox
-        self._skybox = None
 
         # Call super init
         super().__init__()
 
+        # Prepare the initialization dicts
+        self._init_info = {}
+        self._init_state = {}
+        self._init_systems = []
+        self._task_metadata = None
+        self._init_objs = {}
+
+        # If we have any scene file specified, use it to create the objects within it
+        if self.scene_file is not None:
+            # Grab objects info from the scene file
+            with open(self.scene_file, "r") as f:
+                scene_info = json.load(f)
+            self._init_info = scene_info["objects_info"]["init_info"]
+            self._init_state = scene_info["state"]["object_registry"]
+            self._init_systems = list(scene_info["state"]["system_registry"].keys())
+            self._task_metadata = {}
+            try:
+                self._task_metadata = scene_info["metadata"]["task"]
+            except:
+                pass
```

Comment:
why try except?

Reply:
Change this to check if ["metadata"]["task"] exists (it doesn't have to)